### ISSUE TITLE
MQTT Autodiscover Thermostat missing switch

### ISSUE THREAD

**Issue by tfarray** [domoticz/domoticz#6052]
In domoticz, I am missing one entry for my Thermostat in the MQTT Autodiscovered devices.

Device : Portable Reporting Controller Thermostat HVAC	Horstmann (Secure Meters) SRT-321 Thermostat-HVAC
Missing switch : When the Thermostat changes to 'heat' (because it's too cold, and it turns heat on)
I can see the entry posted in MQTT, here is an example of the value : 
mqtt -t >> 17:02:55 {"time":1712156575702,"value":0} zwave/MyThermostat/thermostat_operating_state/endpoint_0/state 

This gives interesting data (not critical) to show when the thermostat wants to turn heat on and off.
Let me know if you want more details

**Comment by waltervl** [comment]
There is some info missing. For instructions how to report mqtt auto discover  issues check the wiki page https://www.domoticz.com/wiki/MQTT_AD_Report_Problems

**Comment by tfarray** [comment]
Hello, sorry for not having followed those instructions, I missed to read that prior to posting. I'm new to MQTT, and I still am learning eveyday.

Config Topic
```
homeassistant/climate/nodeID_5/climate/config
```

Config Payload
```
{
  "min_temp": 5,
  "max_temp": 40,
  "temp_step": 0.5,
  "modes": [
    "heat"
  ],
  "mode_state_template": "{{ {1: \"heat\"}[value_json.value] | default('off') }}",
  "current_temperature_topic": "zwave/nodeID_5/sensor_multilevel/endpoint_0/Air_temperature",
  "current_temperature_template": "{{ value_json.value }}",
  "temperature_state_topic": "zwave/nodeID_5/thermostat_setpoint/endpoint_0/setpoint/1",
  "temperature_state_template": "{{ value_json.value }}",
  "temperature_command_topic": "zwave/nodeID_5/thermostat_setpoint/endpoint_0/setpoint/1/set",
  "mode_state_topic": "zwave/nodeID_5/thermostat_mode/endpoint_0/mode",
  "mode_command_topic": "zwave/nodeID_5/thermostat_mode/endpoint_0/mode/set",
  "action_topic": "zwave/nodeID_5/thermostat_operating_state/endpoint_0/state",
  "action_template": "{{ {0: \"idle\", 1: \"heating\", 2: \"cooling\", 3: \"fan\", 4: \"idle\", 5: \"idle\", 6: \"fan\", 7: \"heating\", 8: \"heating\", 9: \"cooling\", 10: \"heating\", 11: \"heating\"}[value_json.value] | default('idle') }}",
  "temperature_unit": "C",
  "precision": 0.1,
  "device": {
    "identifiers": [
      "zwavejs2mqtt_0xf5c19b66_node5"
    ],
    "manufacturer": "Secure Meters (UK) Ltd.",
    "model": "Battery Powered Wall Thermostat (SRT321)",
    "name": "nodeID_5",
    "sw_version": "1.0"
  },
  "availability": [
    {
      "payload_available": "true",
      "payload_not_available": "false",
      "topic": "zwave/nodeID_5/status",
      "value_template": "{{'true' if value_json.value else 'false'}}"
    },
    {
      "topic": "zwave/_CLIENTS/ZWAVE_GATEWAY-zwave-js-ui/status",
      "value_template": "{{'online' if value_json.value else 'offline'}}"
    },
    {
      "payload_available": "true",
      "payload_not_available": "false",
      "topic": "zwave/driver/status"
    }
  ],
  "availability_mode": "all",
  "name": "nodeID_5_climate",
  "unique_id": "zwavejs2mqtt_0xf5c19b66_Node5_climate"
}
```

State Topic & payload
```
Topic :  zwave/nodeID_5/thermostat_operating_state/endpoint_0/state
Value : {"time":1712216131009,"value":1}
```

But I don't think this is a "real" issue. It's just that the data does not get parsed to Domoticz. There is probably some filters set somewhere, but I don't know if it's per device or if there is a global schema.

**Comment by gizmocuz** [comment]
I get one switch with this configuration

![image](https://github.com/domoticz/domoticz/assets/2265299/d0f11eda-4501-467b-9a88-3368a4d3c4fe)

I think one of the reasons is that 'modes' only contain 'heat' ... no off or other values

**Comment by tfarray** [comment]
Yes, I have this one too, but it's not the good one. On the device, when the heating icon turns on and off, this switch does not turn on/off. But when you look at the MQTT traffic, you can see that the mentioned topic does turn on/off together with the device.

**Comment by gizmocuz** [comment]
I know, and the reason could be very well the 'modes' field in the config as it contains only one mode

**Comment by tfarray** [comment]
This is beyond my understanding/knowledge. If you have any documentation related to this, I would be glad to take time and read it.

**Comment by gizmocuz** [comment]
You could have a look in the MQTTAutoDiscovery.cpp source code.
When using Visual Studio Community, you could set break points when the config is parsed.
You could also read the HASS MQTT Auto Discovery documentation to see what the 'modes' field is for

**Comment by waltervl** [comment]
@gizmocuz When I look at the config payload I read
  "mode_state_template": "{{ {1: \"heat\"}[value_json.value] | default('off') }}",

This would suggest to me that there should be a selector switch be made with "heat" and "off" values....

@tfarray If you look at the Domoticz switch and click on edit button, is the Off value visible? Is the Off value not hidden from the widget (that is a setting for Domoticz Selector switches)

**Comment by gizmocuz** [comment]
@waltervl , no this is a 'template' to translate the values. It should be in the 'modes' array

**Comment by waltervl** [comment]
Fixed in 2025.1 beta? As this seems to be similar to https://github.com/domoticz/domoticz/issues/6309
@tfarray Can you update to latest beta and zigbee2mqtt and retry?

**Comment by tfarray** [comment]
Hello, I have started to use Home Assistant as Primary tool for my home automation, but I still have my domoticz instance running. Since you took time to answer and look at this issue, I also took the time to verify.
I have updated my Domoticz to version 2025.1
And I my domoticz logs I don't see any change on the heat.

This is a not a zigbee device, and I am using Z-Wave JS

Running Version: 2025.1

Domoticz logs : I don't see anything. I can see that I do change my setpoint, but no changes are made for the `Operating State`
```
2025-06-18 17:08:29.211 MQTT Autodiscover: Setpoint/SetPoint ([XX] Thermostat - SetPoint)
2025-06-18 17:08:37.009 MQTT Autodiscover: Setpoint/SetPoint ([XX] Thermostat - SetPoint)
2025-06-18 17:08:38.845 MQTT Autodiscover: Setpoint/SetPoint ([XX] Thermostat - SetPoint)
2025-06-18 17:08:40.841 MQTT Autodiscover: Setpoint/SetPoint ([XX] Thermostat - SetPoint)
2025-06-18 17:08:42.042 MQTT Autodiscover: Setpoint/SetPoint ([XX] Thermostat - SetPoint)
```

 Z-Wave JS : In the GUI, the value does change. Here is the related log with newvalue = 1 (Time does not match I know, I triggered multiple times)
```
6/18/2025, 5:20:12 PM - value updated
Arg 0:
└─commandClassName: Thermostat Operating State
└─commandClass: 66
└─property: state
└─endpoint: 0
└─newValue: 1
└─prevValue: 0
└─propertyName: state
```

In my MQTT, I did also see the change : 
```
zwave/XX/Thermostat/thermostat_operating_state/endpoint_0/state
 {"time":1750260919627,"value":1}
 {"time":1750260874368,"value":0}
```

I am glad to help, and I can do any testing for you if you need to. But personally, this issue can be closed as I am not using anymore domoticz (even if I still run it).

Thank you

**Comment by gizmocuz** [comment]
Thank you for your feedback!

As it is not in the modes table, it will not work. This is probably a config issue in the ZWaveJS side.

HASS is not using MQTT-AD but uses ZWaveJS directly, and has special rules

I like to keep the MQTT-AD part supported as more devices are using this

Thanks for your time, closing the issue.

Personally I can not get used to HASS and it's GUI and lack of scripting. I am trying it every few weeks but get stuck duo some automation issues.
Some user use Domoticz for this, or Node-Red to bypass this... Something like dzVents would be nice.

**Comment by tfarray** [comment]
I am using Node-Red to replace dzvents. The learning curve is easier but dzvents is indeed more powerfull as much more flexible.
